### PR TITLE
Utilize event type for title generation

### PR DESCRIPTION
**Pull request by dcramer**
- Add {Group,Event}.get_event_type()
- Add {Group,Event}.get_event_metadata()
- Deprecate {Group,Event}.message_short
- Deprecate {Group,Event}.error()
- Deprecate {Group,Event}.has_two_part_message()
- Unify default metadata generation for legacy events
- Update {Group,Event}.title to use EventType.to_string()
- Update email subjects to use title

### CODE REVIEW

File: src/sentry/models/event.py [hunk 1]
```diff
@@ -89,27 +89,51 @@ def get_legacy_message(self):
...
+
+    def get_event_metadata(self):
+        """
+        Return the metadata of this event.
+
+        See ``sentry.eventtypes``.
+        """
+        etype = self.data.get('type')
+        if etype is None:
+            etype = 'default'
```
Comment (tkaemming) [inline src/sentry/models/event.py]: Was there a reason not to just use `self.data.get('type', 'default')` here?

File: src/sentry/models/event.py [hunk 2]
```diff
@@ -89,27 +89,51 @@ def get_legacy_message(self):
...
-            message = '<unlabeled message>'
-        else:
-            message = truncatechars(message.splitlines()[0], 100)
-        return message
+        warnings.warn('Event.message_short is deprecated, use Event.title',
+                      DeprecationWarning)
+        return self.title
+
+    def has_two_part_message(self):
+        warnings.warn('Event.message_short is no longer used',
```
Comment (tkaemming) [inline src/sentry/models/event.py]: Wrong method name in copy/paste here.

File: src/sentry/models/group.py [hunk 3]
```diff
@@ -373,29 +374,50 @@ def get_tags(self, with_internal=True):
...
-            message = '<unlabeled message>'
-        else:
-            message = truncatechars(message.splitlines()[0], 100)
-        return message
+        warnings.warn('Group.message_short is deprecated, use Group.title',
+                      DeprecationWarning)
+        return self.title
+
+    def has_two_part_message(self):
+        warnings.warn('Group.message_short is no longer used',
```
Comment (tkaemming) [inline src/sentry/models/group.py]: Also wrong method copy/paste here.

File: src/sentry/utils/javascript.py [hunk 4]
```diff
@@ -185,7 +185,7 @@ def transform(self, obj, request=None):
             'id': six.text_type(obj.id),
             'count': six.text_type(obj.times_seen),
             'title': escape(obj.title),
-            'message': escape(obj.message_short),
+            'message': escape(obj.get_legacy_message()),
```
Comment (tkaemming) [inline src/sentry/utils/javascript.py]: Why'd this method change?

**Review by tkaemming — APPROVED** [review]
A few minor questions inline, but overall assuming you know this better than I do.

File: src/sentry/models/event.py (hunk 1 above)
Comment (dcramer) [inline src/sentry/models/event.py]: not really, I think I had this code doing something else before

File: src/sentry/utils/javascript.py (hunk 4 above)
Comment (dcramer) [inline src/sentry/utils/javascript.py]: it was wrong to begin with. I actually plan to follow up and remove this utility as its super old and not kept up to date.

**Review by tkaemming — APPROVED** [review]
👍

**Pull request merged**